src/provider.ts

```diff
@@ -70,15 +72,34 @@ export class VerifyingProvider {
     blockNumber: bigint | number,
     blockHash: Bytes32,
     chain: bigint | Chain = Chain.Mainnet,
+    customCrypto: CustomCrypto,
   ) {
     this.rpc = new RPC({ URL: providerURL });
-    this.common = new Common({
-      chain,
-      hardfork: chain === Chain.Mainnet ? Hardfork.Shanghai : undefined,
-    });
     const _blockNumber = BigInt(blockNumber);
     this.latestBlockNumber = _blockNumber;
     this.blockHashes[bigIntToHex(_blockNumber)] = blockHash;
+    this.common = new Common({
+      chain,
+      hardfork: chain === Chain.Mainnet ? Hardfork.Cancun : undefined,
```

Comment:
Why not always set the hard fork to `Cancun` instead of setting it to undefined?

Reply:
```suggestion
      hardfork: Hardfork.Cancun,
```